napari/components/viewer_model.py

```diff
@@ -39,47 +44,32 @@ class ViewerModel(AddLayersMixin, KeymapHandler, KeymapProvider):
         Preset color palettes.
     """
 
-    themes = palettes
-
-    def __init__(
-        self, title='napari', ndisplay=2, order=None, axis_labels=None
-    ):
-        super().__init__()
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            status=Event,
-            help=Event,
-            title=Event,
-            interactive=Event,
-            cursor=Event,
-            reset_view=Event,
-            active_layer=Event,
-            palette=Event,
-            grid=Event,
-            layers_change=Event,
-        )
-
+    title: str = 'napari'
+    status: str = 'Ready'
+    help: str = ''
+    cursor: str = 'standard'
+    cursor_size: Optional[int] = None
+    interactive: bool = True
+    active_layer: Optional[int] = None
+    grid_size: Tuple[int, int] = (1, 1)
+    theme: str = DEFAULT_THEME
+
+    grid_stride: ClassVar[int] = 1
+    palette: ClassVar[Dict[str, str]] = palettes[DEFAULT_THEME]
+    layers: ClassVar[LayerList] = []
```

Comment:
`ClassVar` are just treated as normal variables by the dataclass, and won't get their own event emission, which is quite nice https://docs.python.org/3/library/dataclasses.html#class-variables

Reply:
yeah, I'm not totally in love with that pattern... 
Technically, according to [PEP526](https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations):

> `ClassVar` is a special class defined by the typing module that indicates to the static type checker that this variable should not be set on instances.

so while that has no runtime effect, and it _does_ have the desired effect of making these "private" attributes, it's not exactly _semantically_ correct.  So in the commit I just pushed, I added another syntax for excluding events, that is probably better:
```python
layers: LayerList = field(default_factory=LayerList, metadata={'events': False})

# note also that using mutables as default values like I did there is a NO-NO
layers: ClassVar[LayerList] = []
```

Reply:
hmm yeah, i guess in a couple places we could use the `field` syntax or not, and ultimately we'll probably want to make one decision for things to be as consistent as possible ....

Reply:
we're not going to be able to avoid using `field` entirely since it's the only way to provide a mutable default.  For instance, if you want something to start as an empty list you have to use `default_factory=list` (That's a good thing since doing `__init__(self, x = [])` is a bad thing®).  And field is a central player in the dataclass API... so I think also using them to sprinkle metadata here or there in unique cases is fine.